packages/cactus-plugin-bungee-hermes/src/main/typescript/strategy/strategy-ethereum.ts

```diff
@@ -0,0 +1,228 @@
+  size?: number;
+  gasLimit?: number;
+  gasUsed?: number;
+  timestamp?: string;
+  transactions?: Array<unknown>;
+  uncles?: Array<unknown>;
+}
+export interface EthereumNetworkDetails extends NetworkDetails {
+  signingCredential: Web3SigningCredential;
+  keychainId: string;
+  contractName: string;
+  contractAddress: string;
+}
+export class StrategyEthereum implements ObtainLedgerStrategy {
+  public static readonly CLASS_NAME = "StrategyEthereum";
+
+  public log: Logger;
+
+  constructor(level: LogLevelDesc) {
+    this.log = LoggerProvider.getOrCreate({
+      label: StrategyEthereum.CLASS_NAME,
+      level,
+    });
+  }
+
+  public async generateLedgerStates(
+    stateIds: string[],
+    networkDetails: EthereumNetworkDetails,
+  ): Promise<Map<string, State>> {
+    this.log.debug(`Generating ledger snapshot`);
+    const config = new Configuration({
+      basePath: networkDetails.connectorApiPath,
+    });
+    const ethereumApi = new EthereumApi(config);
+
+    const ledgerStates = new Map<string, State>();
+    const assetsKey =
+      stateIds.length == 0
+        ? ((await this.getAllAssetsKey(
+            networkDetails,
+            ethereumApi,
+          )) as string[])
+        : stateIds;
+    this.log.debug("Current assets detected to capture: " + assetsKey);
+    for (const assetKey of assetsKey) {
+      const { transactions, values, blocks } = await this.getAllInfoByKey(
+        assetKey,
+        networkDetails,
+        ethereumApi,
+      );
+
+      const state = new State(assetKey, values, transactions);
+
+      const stateProof = new StateProof(
+        state.getValue(),
+        parseInt(state.getVersion()),
+        state.getId(),
+      );
+      const blocksHash: string[] = [];
+      for (const block of blocks.values()) {
+        if (blocksHash.indexOf(block.hash as string) !== -1) {
+          continue;
+        }
+        blocksHash.push(block.hash as string);
+        stateProof.addBlock({
+          blockHash: block.hash as string,
+          blockCreator: block.miner as string,
+          blockSigners: [], // Non applicable for ethereum
+        });
+      }
+      state.setStateProof([]);
+      ledgerStates.set(assetKey, state);
+    }
+    return ledgerStates;
+  }
+
+  async getAllAssetsKey(
+    networkDetails: EthereumNetworkDetails,
+    api: EthereumApi,
+  ): Promise<string | string[]> {
+    const parameters = {
+      contract: {
+        contractName: networkDetails.contractName,
+        keychainId: networkDetails.keychainId,
+      },
+      //keychainId: networkDetails.keychainId,
+      invocationType: EthContractInvocationType.Call,
+      methodName: "getAllAssetsIDs",
+      params: [],
+      signingCredential: networkDetails.signingCredential,
+    };
+    const response = await api.invokeContractV1(
+      parameters as InvokeContractV1Request,
+    );
+    if (response != undefined) {
+      return response.data.callOutput;
+    }
+
+    return "response undefined";
```

Comment:
changed it to empty array. Also changed the function return type.

The same code was present in the Besu strategy, so i changed that as well

Reply:
@eduv09 Is there any version of events where the `callOutput` being falsy is a valid outcome? I'm guessing not and if that is the case then I recommend throwing an error instead. This prevents domino effect type of bugs that manifest somewhere else that we'll then have to trace back somehow to this point in the code (where it all started going wrong). If you just throw instead then it will fail at the earliest known point when something was going sideways. They call this principle "fail fast".